velox/docs/develop/connectors.rst

```diff
@@ -0,0 +1,91 @@
+==========
+Connectors
+==========
+
+Connectors allow to read and write data to and from external sources.
+This concept is similar to `Presto Connectors <https://prestodb.io/docs/current/develop/connectors.html>`_.
+The :ref:`TableScanNode<TableScanNode>` operator reads external data via a connector.
+The :ref:`TableWriteNode<TableWriteNode>` operator writes data externally via a connector.
+The various connector interfaces in Velox are described below.
+
+Connector Interface
+-------------------
+
+.. list-table::
+   :widths: 10 40
+   :header-rows: 1
+
+   * - Interface Name
+     - Description
+   * - ConnectorSplit
+     - Basic unit that describes a chunk of data for processing.
+   * - DataSource
+     - Provides methods to consume and process a split. A DataSource can optionally consume a
+       dynamic filter during execution to prune some rows from the output vector.
+   * - DataSink
+     - Provides methods to write a Velox vector externally.
+   * - Connector
+     - Allows creating instances of a DataSource or a DataSink.
+   * - Connector Factory
+     - Enables creating instances of a particular connector.
+
+Velox currently has in-built support for the Hive Connector and the TPC-H Connector.
+Let's see how the above connector interfaces are implemented in the Hive Connector in detail below.
+
+Hive Connector
+--------------
+The Hive Connector can be used to read and write data files (Parquet, DWRF) residing on
+an external storage (S3, HDFS, GCS, Linux FS).
+
+HiveConnectorSplit
+~~~~~~~~~~~~~~~~~~
+The HiveConnectorSplit describes a data chunk using parameters including `file-path`,
```

Comment:
Maybe clarify how a chunk of file is processed. Many readers would wonder whether they have to specify offset and length that align with rows It would be nice to explain that this is not necessary and how the split is processed when offsets doesn't land on a row boundary.